File: burncup-fe/src/model/competition_model.ts
export interface Competition {
    id: string;
    name: string;
    description: string;
    category: string;
    imageUrl: string;
    bookletUrl: string;
    paidMessage: string;
    registrationStartDate: string;
    registrationEndDate: string;
    competitionStartDate: string;
    competitionEndDate: string;
    competitionType: string; // e.g., "Binusian", "NonBinusian", "Mixed"
    venue: string;
    registrationfee: number;
    prizes: Prize[];
    requirements: string[];
    rules: string[];
    maxMembers?: number | null; // Optional, for team competitions
    minMembers?: number | null; // Optional, for team competitions
    teamSlot: number; // Number of teams that can register
    createdAt: string;
    updatedAt: string;
}

export interface Prize {
    id: string;
    name: string;
    description: string;
}

export interface CompetitionStats {
  id: string
  name: string
  category: string
  totalTeams: number
  totalParticipants: number
  paidTeams: number
  pendingTeams: number
  registrationFee: number
  competitionType: string
}

export interface CompetitionCarouselProps {
  competitions: CompetitionStats[]
}

File: burncup-be/models/competition_model.go
package models

// Prize struct matches the Prize interface and DB schema
type Prize struct {
	ID          string `db:"id" json:"id"`
	Name        string `db:"name" json:"name"`
	Description string `db:"description" json:"description"`
}

// Competition struct matches the Competition interface and DB schema
type Competition struct {
	ID                    string   `db:"id" json:"id"`
	Name                  string   `db:"name" json:"name"`
	Description           string   `db:"description" json:"description"`
	Category              string   `db:"category" json:"category"`
	ImageURL              string   `db:"image_url" json:"imageUrl"`
	BookletURL            string   `db:"booklet_url" json:"bookletUrl"`
	PaidMessage           string   `db:"paid_message" json:"paidMessage"`
	RegistrationStartDate string   `db:"registration_start_date" json:"registrationStartDate"`
	RegistrationEndDate   string   `db:"registration_end_date" json:"registrationEndDate"`
	CompetitionStartDate  string   `db:"competition_start_date" json:"competitionStartDate"`
	CompetitionEndDate    string   `db:"competition_end_date" json:"competitionEndDate"`
	CompetitionType       string   `db:"competition_type" json:"competitionType"`
	Venue                 string   `db:"venue" json:"venue"`
	RegistrationFee       int      `db:"registration_fee" json:"registrationfee"`
	MaxMembers            *int     `db:"max_members" json:"maxMembers,omitempty"`
	MinMembers            *int     `db:"min_members" json:"minMembers,omitempty"`
	TeamSlot              int      `db:"team_slot" json:"teamSlot"`
	CreatedAt             string   `db:"created_at" json:"createdAt"`
	UpdatedAt             string   `db:"updated_at" json:"updatedAt"`
	Prizes                []Prize  `json:"prizes"`
	Requirements          []string `json:"requirements"`
	Rules                 []string `json:"rules"`
}
